llama-index-core/llama_index/core/memory/simple_composable_memory.py

```diff
@@ -0,0 +1,141 @@
+from typing import Any, List, Optional
+
+from llama_index.core.base.llms.types import ChatMessage, MessageRole
+from llama_index.core.bridge.pydantic import PrivateAttr
+from llama_index.core.memory.types import (
+    BaseComposableMemory,
+    BaseMemory,
+)
+from llama_index.core.memory import ChatMemoryBuffer
+
+DEFAULT_INTRO_HISTORY_MESSAGE = "Below are a set of relevant dialogues retrieved from potentially several memory sources:"
+DEFAULT_OUTRO_HISTORY_MESSAGE = "This is the end of the of retrieved message dialogues."
+
+
+class SimpleComposableMemory(BaseComposableMemory):
+    """A simple composition of potentially several memory sources.
+
+    This composable memory considers one of the memory sources as the main
+    one and the others as secondary. The secondary memory sources get added to
+    the chat history only in either the system prompt or to the first user
+    message within the chat history.
+    """
+
+    _primary_memory: BaseMemory = PrivateAttr()
+    _secondary_memory_sources: List[BaseMemory] = PrivateAttr()
+
+    def __init__(self, sources: List[BaseMemory]) -> None:
+        if len(sources) == 0:
+            raise ValueError("Must supply at least one memory source.")
+
+        self._primary_memory = sources[0]
+        self._secondary_memory_sources = sources[1:]
+        super().__init__(sources=sources)
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "SimpleComposableMemory"
+
+    @classmethod
+    def from_defaults(
+        cls,
+        sources: Optional[List[BaseMemory]] = None,
+    ) -> "SimpleComposableMemory":
+        """Create a simple composable memory from an LLM."""
+        sources = sources or [ChatMemoryBuffer.from_defaults()]
+
+        return cls(sources=sources)
+
+    def _format_secondary_messages(
+        self, secondary_chat_histories: List[List[ChatMessage]]
+    ) -> str:
+        """Formats retrieved historical messages into a single string."""
+        formatted_history = "\n\n" + DEFAULT_INTRO_HISTORY_MESSAGE + "\n"
```

Comment:
this could probably be a prompt template? maybe? (imo probably a future TODO, but just a note)

Reply:
ahh very true.. going to note it as a future TODO (I might crush tmr or the next day lol)